src/objects.c

```diff
@@ -1799,3 +1813,748 @@ int p11prov_obj_export_public_ec_key(P11PROV_OBJ *obj, OSSL_CALLBACK *cb_fn,
+                                  void *cb_arg)
+{
+    if (obj == NULL) {
+        return RET_OSSL_ERR;
+    }
+
+    if (obj->class != CKO_PRIVATE_KEY && obj->class != CKO_PUBLIC_KEY) {
+        P11PROV_raise(obj->ctx, CKR_GENERAL_ERROR, "Invalid Object Class");
+        return RET_OSSL_ERR;
+    }
+
+    if (key_type != CK_UNAVAILABLE_INFORMATION) {
+        if (key_type != obj->data.key.type) {
+            P11PROV_raise(obj->ctx, CKR_GENERAL_ERROR, "Invalid Key Type");
+            return RET_OSSL_ERR;
+        }
+    }
+
+    if (!search_related && obj->class != CKO_PUBLIC_KEY) {
+        P11PROV_raise(obj->ctx, CKR_GENERAL_ERROR, "Not a public Key");
+        return RET_OSSL_ERR;
+    }
+
+    switch (obj->data.key.type) {
+    case CKK_RSA:
+        return p11prov_obj_export_public_rsa_key(obj, cb_fn, cb_arg);
+    case CKK_EC:
+    case CKK_EC_EDWARDS:
+        return p11prov_obj_export_public_ec_key(obj, cb_fn, cb_arg);
+    default:
+        P11PROV_raise(obj->ctx, CKR_GENERAL_ERROR, "Unsupported key type");
+        return RET_OSSL_ERR;
+    }
+}
+
+int p11prov_obj_get_ec_public_x_y(P11PROV_OBJ *obj, CK_ATTRIBUTE **pub_x,
+                                  CK_ATTRIBUTE **pub_y)
+{
+    const unsigned char *val;
+    void *tmp_ptr;
+    CK_ATTRIBUTE *ec_params;
+    CK_ATTRIBUTE *pub_key;
+    EC_POINT *pub_point = NULL;
+    EC_GROUP *group = NULL;
+    CK_ATTRIBUTE *a_x;
+    CK_ATTRIBUTE *a_y;
+    BN_CTX *bnctx = NULL;
+    BIGNUM *x;
+    BIGNUM *y;
+    int len;
+    int ret;
+
+    if (!obj) {
+        return RET_OSSL_ERR;
+    }
+
+    if (obj->class != CKO_PRIVATE_KEY && obj->class != CKO_PUBLIC_KEY) {
+        P11PROV_raise(obj->ctx, CKR_GENERAL_ERROR, "Invalid Object Class");
+        return RET_OSSL_ERR;
+    }
+
+    if (obj->data.key.type != CKK_EC) {
+        P11PROV_raise(obj->ctx, CKR_GENERAL_ERROR, "Unsupported key type");
+        return RET_OSSL_ERR;
+    }
+
+    /* See if we have cached attributes first */
+    a_x = p11prov_obj_get_attr(obj, CKA_P11PROV_PUB_KEY_X);
+    a_y = p11prov_obj_get_attr(obj, CKA_P11PROV_PUB_KEY_Y);
+    if (a_x && a_y) {
+        if (pub_x) {
+            *pub_x = a_x;
+        }
+        if (pub_y) {
+            *pub_y = a_y;
+        }
+        return RET_OSSL_OK;
+    }
+
+    ec_params = p11prov_obj_get_attr(obj, CKA_EC_PARAMS);
+    if (!ec_params) {
+        return RET_OSSL_ERR;
+    }
+    pub_key = p11prov_obj_get_attr(obj, CKA_P11PROV_PUB_KEY);
+    if (!pub_key) {
+        return RET_OSSL_ERR;
+    }
+
+    bnctx = BN_CTX_new();
+    if (!bnctx) {
+        ret = RET_OSSL_ERR;
+        goto done;
+    }
+    /* prevent modification of the attribute pointer */
+    val = ec_params->pValue;
+    group = d2i_ECPKParameters(NULL, &val, ec_params->ulValueLen);
+    if (!group) {
+        ret = RET_OSSL_ERR;
+        goto done;
+    }
+
+    x = BN_CTX_get(bnctx);
+    y = BN_CTX_get(bnctx);
+    if (!y) {
```

Comment:
missing null check for `x`.

Reply:
From openssl man page directly:
`Once BN_CTX_get() has failed, the subsequent calls will return NULL as well, so it is sufficient to check the return value of the last BN_CTX_get() call.`